index.js

```diff
@@ -157,15 +192,7 @@ function test(example, value) {
     }
 
     if (example.regexp) {
-      var regexp;
-      if (example.regexp instanceof RegExp) {
-        regexp = example.regexp;
-      } else if (typeof example.regexp === 'string') {
-        regexp = toRegExp(example.regexp);
-      } else {
-        throw new TypeError('Invalid regular expression passed to regexp query.');
-      }
-      return regexp.test(value);
+      return value ? value.match(RegExp(example.regexp)) : false;
```

Comment:
Isn't this a breaking change too? I mean before your change, when the `regexp` value was not a string or RegExp instance, the code thrown a reasonably helpful error. With your change in place, we will try to convert anything to a regexp, e.g. `RegExp(0)` or `RegExp(new Date())`.

I think we should preserve the old code.

What are your arguments for making this change?